lib/src/builder/codegen/accessors_generator.dart

```diff
@@ -308,9 +308,78 @@ abstract class TypedMapAccessorsGenerator extends BoilerplateDeclarationGenerato
+        var setter = convertProp?.arguments?.arguments.firstOrNull?.toSource();
+        var getter = convertProp?.arguments?.arguments.lastOrNull?.toSource();
+
+        UnsupportedError buildPropAnnotationError(String message) {
+          return UnsupportedError(
+              'Unsupported prop annotation combination for prop $accessorName: $message');
+        }
+
+        if (convertProp != null) {
+          if (rawType == null || convertedType == null) {
+            throw buildPropAnnotationError(
+                'The @ConvertProp annotation must be used with generic parameters: `@Convert<Raw, Converted>(setter, getter)`');
+          }
+          if (setter == null || getter == null) {
+            // Analysis errors with `ConvertProp` should prevent this from happening.
+            throw buildPropAnnotationError(
+                'The @ConvertProp annotation must have two arguments: `@Convert<Raw, Converted>(setter, getter)`');
+          }
+          if (convertedType != typeSource) {
+            throw buildPropAnnotationError(
+                'A prop annotated with `@Convert<Raw, Converted>(setter, getter)` should have the same type as the `Converted` generic parameter.');
+          }
+          shouldConvertProp = true;
+        }
+
+        // Look for special-case conversion annotations (@convertJsMapProp and
+        // @convertJsRefProp) only if @ConvertProp(...) annotation isn't already used.
+        if (!shouldConvertProp) {
+          final convertJsMapProp =
+              getConstantAnnotation(field, 'convertJsMapProp', annotations.convertJsMapProp);
+          final convertJsRefProp =
+              getConstantAnnotation(field, 'convertJsRefProp', annotations.convertJsRefProp);
+          if (convertJsMapProp != null) {
+            rawType = 'JsMap?';
+            convertedType = 'Map?';
+            setter = 'jsifyMapProp';
+            getter = 'unjsifyMapProp';
+            shouldConvertProp = true;
+            if (convertedType != typeSource) {
+              throw buildPropAnnotationError(
+                  'A prop annotated with `@convertJsMapProp` should be typed as `Map?`.');
+            }
+          } else if (convertJsRefProp != null) {
+            rawType = 'dynamic';
+            convertedType = 'dynamic';
+            setter = 'jsifyRefProp';
+            getter = 'unjsifyRefProp';
+            shouldConvertProp = true;
+            if (convertedType != typeSource) {
+              throw buildPropAnnotationError(
+                  'A prop annotated with `@convertJsRefProp` should be typed as `dynamic`.');
+            }
+          }
+        }
+
         String castGetterMapValueIfNecessary(String expression) {
           if (typeSource == null) return expression;
-          return '($expression) as $typeSource';
+          return '($expression) as ${shouldConvertProp ? rawType : typeSource}';
+        }
+
+        String convertIfNecessary(String expression, [bool isGetter = true]) {
```

Comment:
Could we make this a named arg, so it's clear what the boolean represents (see [this Dart style guide note](https://dart.dev/effective-dart/design#avoid-positional-boolean-parameters) about positional booleans), and also required to ensure callers are explicit about which they want?